game-core/src/main/java/games/strategy/triplea/oddsCalculator/ta/AggregateResults.java

```diff
@@ -41,42 +41,32 @@ public void addResults(final Collection<BattleResults> results) {
     return results;
   }
 
-  /**
-   * @return {@code null} if we have zero results.
-   */
-  public @Nullable BattleResults getBattleResultsClosestToAverage() {
-    double closestBattleDif = Integer.MAX_VALUE;
-    BattleResults closestBattle = null;
-    for (final BattleResults result : results) {
-      double dif = Math.abs(result.getAttackingCombatUnitsLeft() - getAverageAttackingUnitsLeft());
-      dif += Math.abs(result.getDefendingCombatUnitsLeft() - getAverageDefendingUnitsLeft());
-      if (dif < closestBattleDif) {
-        closestBattleDif = dif;
-        closestBattle = result;
-      }
-    }
-    return closestBattle;
+  private Optional<BattleResults> getBattleResultsClosestToAverage() {
+    return results.stream()
+        .min(Comparator.comparingDouble(
+            result -> Math.abs(result.getAttackingCombatUnitsLeft() - getAverageAttackingUnitsLeft())
+                + Math.abs(result.getDefendingCombatUnitsLeft() - getAverageDefendingUnitsLeft())));
   }
 
   public List<Unit> getAverageAttackingUnitsRemaining() {
-    final BattleResults result = getBattleResultsClosestToAverage();
-    return result == null ? new ArrayList<>() : result.getRemainingAttackingUnits();
+    return getBattleResultsClosestToAverage()
+        .map(BattleResults::getRemainingAttackingUnits)
+        .orElseGet(ArrayList::new);
   }
 
   public List<Unit> getAverageDefendingUnitsRemaining() {
-    final BattleResults result = getBattleResultsClosestToAverage();
-    return result == null ? new ArrayList<>() : result.getRemainingDefendingUnits();
+    return getBattleResultsClosestToAverage()
+        .map(BattleResults::getRemainingDefendingUnits)
+        .orElseGet(ArrayList::new);
   }
 
   double getAverageAttackingUnitsLeft() {
     if (results.isEmpty()) {
       return 0.0;
     }
-    double count = 0;
-    for (final BattleResults result : results) {
-      count += result.getAttackingCombatUnitsLeft();
-    }
-    return count / results.size();
+    return results.stream()
+        .mapToDouble(BattleResults::getAttackingCombatUnitsLeft)
```

Comment:
For consistency, would you want to change this to `mapToInt()` as you did when using `getDefendingCombatUnitsLeft()` below?  Or change below to `mapToDouble()`?